sphinx/util/requests.py

```diff
@@ -77,3 +77,38 @@ def head(url: str,
 
     with ignore_insecure_warning(verify):
         return requests.head(url, **kwargs)
+
+
+class _Session(requests.Session):
+
+    def get(self, url: str, **kwargs: Any) -> requests.Response:  # type: ignore[override]
+        """Sends a GET request like requests.get().
+
+        This sets up User-Agent header and TLS verification automatically."""
+        headers = kwargs.setdefault('headers', {})
+        headers.setdefault('User-Agent', kwargs.pop('_user_agent', _USER_AGENT))
```

Comment:
The `kwargs.pop` pattern used here - and also for `_tls_info` - might not be ideal.  I began using that due to line length limit complaints by `flake8` with the additional indenting required for class methods compared to a top-level module method.

Reply:
I think I should probably take one more attempt at restoring the type-checking for `_tls_info` and `_user_agent`.  One of those things that could exist for a long time if it isnt'f fixed during this changeset.

Reply:
(done)

Reply:
Ah.. but actually using those parameters would make sense!  Oops.  One moment..

Reply:
OK.  I made a bit of a mess of that, but I think that both of the session-based methods should now be reading the `_user_agent` and `_tls_info` arguments much like the previous, individual-request-based methods were, and with the same level of argument typechecking (at least when used through the `_Session` class).